ext/serializer.c

```diff
@@ -1834,6 +1924,43 @@ zval *ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         }
     }
 
+    // Figure out if an error should be tracked
```

Comment:
The existing code for marking entry spans as error is in dd_set_entrypoint_root_span_props_end.
Marking as error for the client integration should happen in CurlIntegration.php (in set_curl_attributes function).

Reply:
Generally marking as error should be immediate, so that users may manually unmark as error, e.g. in onClose handlers or similar.